### PR TITLE
Lex/fee fetch loop

### PR DESCRIPTION
**Pull request by LPeterson18**
1. Updated Databases to structure around partners
2. Added fetch calls for ethgasstation.info, mempool.space, and bitcoinfees.earn.com
3. Added engine to fetch from APIs and add collected fees to DBs with normalized dates as Ids

### CODE REVIEW

**Comment by LPeterson18** [comment]
Rename and restructure files and folders for better readability

**Comment by LPeterson18** [comment]
Re-organized file structure as detailed during PR review

File: src/feesEngine.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          }
+          mylog(preppedData)
+          await dbEthGS.insert(preppedData)
+        } else {
+          mylog('Error Normalizing Date: Invalid Date Format')
+        }
+      } else if (feesResult[0] === null) {
+        mylog('Error getting eth fees from Eth Gas Station')
+      }
+      if (feesResult[0] != null) {
```
Comment (thehobbit85) [inline src/feesEngine.ts]: This should probably be `feesResult[1]` no?

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comment

**Comment by LPeterson18** [comment]
Fixed error handling

**Comment by LPeterson18** [comment]
Rebased branch

File: src/partners/earn.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export const earnCall = async (): Promise<EarnInfo | null> => {
+  const options = {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json'
+    }
+  }
+  const url = `https://bitcoinfees.earn.com/api/v1/fees/list`
```
Comment (LPeterson18) [inline src/partners/earn.ts]: Base URIs for all providers should be in config file

File: src/feesEngine.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+const snooze = async (ms: number): Promise<void> =>
+  await new Promise((resolve: Function) => setTimeout(resolve, ms))
+
+const feesLoop = async (): Promise<GasInfo | BitcoinInfo | EarnInfo> => {
+  while (true) {
+    let preppedData
+    let prePrepData
+    const earn = earnCall()
+    const mempool = mempoolSpaceCall()
+    const ethGasStation = ethGasStationCall()
```
Comment (LPeterson18) [inline src/feesEngine.ts]: Handle result/rejected promises on each one and not on returned array from promise.all

File: src/feesEngine.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+const nanoDb = nano(config.dbFullpath)
+const dbEthGS = nanoDb.db.use('fees_ethgasstation')
+const dbEarn = nanoDb.db.use('fees_mempoolspace')
+const dbMempool = nanoDb.db.use('fees_earn')
+promisify(dbEthGS)
+promisify(dbEarn)
+promisify(dbMempool)
+
+const snooze = async (ms: number): Promise<void> =>
+  await new Promise((resolve: Function) => setTimeout(resolve, ms))
```
Comment (LPeterson18) [inline src/feesEngine.ts]: Move to utils file

**Comment by LPeterson18** [comment]
1. Moved Base URIs to config
2. Changed engine to handle resolved/rejected promises on each one and not on returned array from promise.all
3. Moved snooze to utils
4. Reorganized commits

**Comment by LPeterson18** [comment]
1. Reduced fees processing logic into one function using cleaners

**Comment by LPeterson18** [comment]
Removed redundant code by creating generic fetch function

**Comment by LPeterson18** [comment]
Fixed eslint-disable-next-line and rebased branch

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**